### PR TITLE
backend/feat: #785 added face verification service support

### PR DESCRIPTION
**Pull request by VD321**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates

## Description
Added face verification support in Verification Module.

### Additional Changes

- [ ] This PR modifies the database schema (database migration added)
- [ ] This PR modifies dhall configs/environment variables

## Checklist

- [ ] I formatted the code and addressed linter errors `./dev/format-all-files.sh`
- [ ] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

File: lib/mobility-core/src/Kernel/External/Verification/Interface/Types.hs [hunk 1]
```diff
@@ -21,9 +21,10 @@ where
 import Deriving.Aeson
 import qualified Kernel.External.Verification.Idfy.Config as Idfy
 import qualified Kernel.External.Verification.Idfy.Types.Response as Idfy
+import qualified Kernel.External.Verification.InternalScripts.Types as FV
 import Kernel.Prelude
 
-newtype VerificationServiceConfig = IdfyConfig Idfy.IdfyCfg
+data VerificationServiceConfig = IdfyConfig Idfy.IdfyCfg | FaceVerificationConfig FV.FaceVerificationCfg
```
Comment (piyushKumar-1) [inline lib/mobility-core/src/Kernel/External/Verification/Interface/Types.hs:27]: i think we should treat these as separate services ? One for DL/RC and other for Facial ? seems mutually exclusive to me.

Comment (VD321) [reply lib/mobility-core/src/Kernel/External/Verification/Interface/Types.hs:27]: @hkmangla asked it to be added as a part of the verification service.

File: lib/mobility-core/src/Kernel/External/Verification/Interface.hs [hunk 2]
```diff
@@ -58,6 +69,20 @@ validateImage ::
   m ValidateImageResp
 validateImage serviceConfig req = case serviceConfig of
   IdfyConfig cfg -> Idfy.validateImage cfg req
+  FaceVerificationConfig _ -> throwError $ InternalError "Not Implemented!"
+
+validateFaceImage ::
+  ( MonadThrow m,
+    CoreMetrics m,
+    MonadIO m,
```
Comment (olegromashin) [inline lib/mobility-core/src/Kernel/External/Verification/Interface.hs:77]: `Kernel.Types.App.MonadFlow` already contains `MonadIO` and `MonadThrow`, hence you can remove them from here.

Comment (olegromashin) [reply lib/mobility-core/src/Kernel/External/Verification/Interface.hs:77]: same for `validateFace` function

File: lib/mobility-core/src/Kernel/External/Verification/Interface.hs [hunk 3]
```diff
@@ -19,15 +19,24 @@ module Kernel.External.Verification.Interface
...
 where
 
+import Control.Monad.Error.Class as Reexport (MonadError)
+import EulerHS.Prelude
+import Kernel.External.Encryption (EncFlow)
 import Kernel.External.Verification.Idfy.Config as Reexport
 import qualified Kernel.External.Verification.Interface.Idfy as Idfy
 import Kernel.External.Verification.Interface.Types as Reexport
+import Kernel.External.Verification.InternalScripts.Error as Reexport
+import qualified Kernel.External.Verification.InternalScripts.FaceVerification as FV
```
Comment (hkmangla) [inline lib/mobility-core/src/Kernel/External/Verification/Interface.hs:33]: You should create a file inside folder `Interface` like Idfy

File: lib/mobility-core/src/Kernel/External/Verification/InternalScripts/Error.hs [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+  | PoorImageQuality
+  deriving (Eq, Show, IsBecknAPIError)
+
+instanceExceptionWithParent 'HTTPException ''FaceVerificationError
+
+instance FromResponse FaceVerificationError where
+  fromResponse resp = case statusCode $ responseStatusCode resp of
+    400 -> Just FakeFaceDetected
+    500 -> Just FaceVerificationInternalServer
+    _ -> Just FaceVerificationInternalServer
```
[the remark this answers is not in the document]
Comment (VD321) [reply lib/mobility-core/src/Kernel/External/Verification/InternalScripts/Error.hs:37]: Changes Error code to 422. Removed case for 500.

**Review by hkmangla — CHANGES_REQUESTED** [review]

**Review by hkmangla — APPROVED** [review]

**Pull request merged**